lib/ChGovUk/Controllers/Company/Document.pm

```diff
@@ -35,13 +35,19 @@ sub document {
             ->get->on(
                 failure => sub {
                     my ($api, $tx) = @_;
+                    my $code = $tx->error->{code} // 0;
 
-                    $delay->emit('error', sprintf(
-                        'Failure fetching filing history item for company_number [%s] filing_history_id [%s] - %s',
-                        $self->stash->{company_number},
-                        $self->stash->{filing_history_id},
-                        $tx->error->{message}
-                    ));
+                    if ($code == 404) {
+                        $delay->emit('not_found', $document_metadata_uri);
```

Comment:
I think since the filing-history get failed we won't have access to a $document_metadata_uri here to pass to the not_found emit